mewarpx/mewarpx/utils_store/profileparser.py

```diff
@@ -0,0 +1,136 @@
+
+            # replace the multiple spaces with a single space
+            line = re.sub("\s+", " ", line)
+            values = line.split(" ")
+
+            name = values[0]
+            n_calls = int(values[1])
+            excl_min = float(values[2])
+            excl_avg = float(values[3])
+            excl_max = float(values[4])
+            max_percent = float(values[5].replace("%", ""))
+
+            self.profile_dict["exclusive_function_profiles"][name] = {
+                "n_calls" : n_calls,
+                "excl_min" : excl_min,
+                "excl_avg" : excl_avg,
+                "excl_max" : excl_max,
+                "max_percent" : max_percent
+            }
+
+            index = index + 1
+
+        # once again delete all the stuff that was already checked plus the next 5 lines
+        del self.lines[0:index + 5]
+
+        for line in self.lines:
+            if "------" in line:
+                break
+
+            line = re.sub("\s+", " ", line)
+            values = line.split(" ")
+
+            name = values[0]
+            n_calls = int(values[1])
+            incl_min = float(values[2])
+            incl_avg = float(values[3])
+            incl_max = float(values[4])
+            max_percent = float(values[5].replace("%", ""))
+
+            self.profile_dict["inclusive_function_profiles"][name] = {
+                "n_calls" : n_calls,
+                "excl_min" : incl_min,
+                "excl_avg" : incl_avg,
+                "excl_max" : incl_max,
+                "max_percent" : max_percent
+            }
+        print("Finished parsing profile output!")
+        self.write_file()
+
+    def write_file(self):
+        if not self.profile_dict:
+            raise RuntimeError("write_file called before profile data was parsed")
+        print("Writing profile data json file...")
+        with open("profile_data.json", "w") as data_file:
+            data_file.write(json.dumps(self.profile_dict, indent=4))
+
+        print("Finished writing json file as 'profile_data.json'")
+
+def main():
+    if len(sys.argv) != 2:
+        raise RuntimeError(f"Usage: {sys.argv[0]} <path to stdout file>")
+
+    path = sys.argv[1]
+
+    print(f"Path to stdout file: {path}")
+
+    output_file = open(path, "r")
+
+    profile_lines = []
+    reached_profile = False
+    print("Parsing stdout file...")
+    count = 0
+    while True:
+        line = output_file.readline()
+
+        if not line:
+            break
+
+        if reached_profile:
+            profile_lines.append(line.replace("\n", ""))
+            count = count + 1
+        elif "TinyProfiler total time across processes" in line:
+            reached_profile = True
+            profile_lines.append(line.replace("\n", ""))
+            count = count + 1
+
+    print(f"Finished parsing stdout file! Found {count} profiling lines")
+
+    full_profile = FullProfile(profile_lines)
+
+    full_profile.parse_and_write_full_profiling_output()
```

Comment:
For the sake of good design, and potential programmatic use that won't go through reading the JSON file in, please just call `full_profile.parse(); full_profile.write_file()` here, and don't have a method `do_this_and_that`

Reply:
One motivation is that Hatchet can be used in scripts, consuming the dictionary directly, rather than having to read in the JSON

Reply:
> For the sake of good design, and potential programmatic use that won't go through reading the JSON file in, please just call full_profile.parse(); full_profile.write_file() here, and don't have a method do_this_and_that

Yep that's much better, thanks.